src/main.rs

```diff
@@ -30,34 +35,64 @@ impl Terminal {
         }
     }
 
-    fn should_ask_for_todo(&mut self) -> bool {
+    fn should_ask_for_todo(&mut self) -> Result<bool, TerminalError> {
         let mut buf = String::new();
-        self.stdin.read_line(&mut buf).unwrap();
 
-        buf.trim() == "s"
+        if let Err(msg_err) = self.stdin.read_line(&mut buf) {
+            self.show_error(TerminalError::StdoutErr(msg_err));
+            std::process::exit(1);         
+        }
+
+        Ok(buf.trim() == "s")
```

Comment:
Aqui o melhor é você retornar o erro ao invés de finalizar o programa. Esse tipo de lógica normalmente fica em uma camada mais superior.

Uma forma de fazer isso é:

```suggestion
        if let Err(msg_err) = self.stdin.read_line(&mut buf) {
            Err(TerminalError::StdoutErr(msg_err))        
        } else {
            Ok(buf.trim() == "s")
        }
```

Mas também podemos usar [`Result::map_err`](https://doc.rust-lang.org/std/result/enum.Result.html#method.map_err) e `?`:

```rust
self.stdin.read_line(&mut buf).map_err(TerminalError::StdoutErr)?;

Ok(buf.trim() == "s")
```

Que acaba sendo a solução mais idiomática.

Reply:
Entendi. Muito obrigado, me tirou uma duvida grande.